src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -117,12 +117,16 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
 
             if (_host != null)
             {
+                int posColon = -1;
+                bool isHostTypeIPv6 = (-1 != (posColon = _host.IndexOf(':')) && -1 != (posColon = _host.IndexOf(':', posColon)));
```

Comment:
Isn't the second _host.IndexOf always going to succeed, because you're passing in posColon rather than posColon+1?  Seems like we need some more tests, too, to catch this kind of thing.

Is it true that every IPv6 address contains at least two colons, and that if an address contains two colons it must be an IPv6 address?  If so, with the aforementioned change, the logic seems fine.

If it's not true, a better approach might be to augment the HttpConnectionPool ctor to accept a "Uri exemplarUri", then at the ctor call site pass in the request.Uri, and here use Uri.HostNameType instead of doing this check manually.

Reply:
Yes, every ipv6 contains at least two colons and nothing else does.

Reply:
Ok, then with the index fix, this looks good, e.g.
```C#
bool isHostTypeIPv6 = (posColon = _host.IndexOf(':')) != -1 && _host.IndexOf(':', posColon+1) != -1;
```

Reply:
Will fix.